glyphs-reader/src/font.rs

```diff
@@ -145,24 +150,48 @@ impl Glyph {
 
 impl FromPlist for Node {
     fn from_plist(plist: Plist) -> Self {
-        let mut spl = plist.as_str().unwrap().splitn(3, ' ');
-        let x = spl.next().unwrap().parse().unwrap();
-        let y = spl.next().unwrap().parse().unwrap();
-        let pt = Point::new(x, y);
-        let node_type = spl.next().unwrap().parse().unwrap();
-        Node { pt, node_type }
+        match &plist {
+            Plist::String(value) => {
+                let mut spl = value.splitn(3, ' ');
+                let x = spl.next().unwrap().parse().unwrap();
+                let y = spl.next().unwrap().parse().unwrap();
+                let pt = Point::new(x, y);
+                let node_type = spl.next().unwrap().parse().unwrap();
+                Node { pt, node_type }
+            }
+            Plist::Array(value) => {
+                if value.len() != 3 {
+                    panic!("Invalid node content {:?}", plist);
+                };
+                let x = value[0].as_f64().unwrap();
+                let y = value[1].as_f64().unwrap();
+                let pt = Point::new(x, y);
+                let node_type = value[2].as_str().unwrap().parse().unwrap();
+                Node { pt, node_type }
+            }
+            _ => {
+                panic!("Invalid node content {:?}", plist);
+            }
+        }
     }
 }
 
 impl std::str::FromStr for NodeType {
     type Err = String;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
+            // Glyphs 2 style
             "LINE" => Ok(NodeType::Line),
             "LINE SMOOTH" => Ok(NodeType::LineSmooth),
             "OFFCURVE" => Ok(NodeType::OffCurve),
             "CURVE" => Ok(NodeType::Curve),
             "CURVE SMOOTH" => Ok(NodeType::CurveSmooth),
+            // Glyphs 3 style
+            "l" => Ok(NodeType::Line),
```

Comment:
same here, you are ignoring format version and matching  either all-cap node types and Glyphs3's shorthands. Even if it may be simpler to do so, I  think it's more correct to distinguish the two code paths

Reply:
I think the same reasoning as https://github.com/googlefonts/fontmake-rs/pull/30/files#r1040811887 applies